### ISSUE TITLE
m2m_analysis : ignore unproducible targets

### ISSUE THREAD

**Issue by cfrioux** [AuReMe/metage2metabo#23]
Running m2m_analysis with a target that is not producible by the community leads to the following error:

```
Traceback (most recent call last):
  File "xxx/miniconda3/envs/m2m/lib/python3.7/logging/__init__.py", line 1025, in emit
    msg = self.format(record)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/logging/__init__.py", line 869, in format
    return fmt.format(record)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/logging/__init__.py", line 608, in format
    record.message = record.getMessage()
  File "xxx/miniconda3/envs/m2m/lib/python3.7/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "xxx/miniconda3/envs/m2m/bin/m2m_analysis", line 8, in <module>
    sys.exit(main())
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/__main_analysis__.py", line 289, in main
    args.oog, new_arg_modelhost, args.level)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/__main_analysis__.py", line 303, in main_analysis_workflow
    run_analysis_workflow(*allargs)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/m2m_analysis/m2m_analysis_workflow.py", line 46, in run_analysis_workflow
    gml_output = graph_analysis(json_file_folder, target_folder_file, output_dir, taxon_file, taxonomy_level)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/m2m_analysis/solution_graph.py", line 58, in graph_analysis
    create_gml(json_paths, target_paths, output_dir, taxonomy_output_file)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/m2m_analysis/solution_graph.py", line 116, in create_gml
    logger.warning('ERROR ', dicti["still_unprod"], ' is unproducible')
Message: 'ERROR '
Arguments: (['M_CPD__45__1823_c', 'M_HISTAMINE_c', 'M_TYRAMINE_c'], ' is unproducible')
--- Logging error ---
Traceback (most recent call last):
  File "xxx/miniconda3/envs/m2m/lib/python3.7/logging/__init__.py", line 1025, in emit
    msg = self.format(record)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/logging/__init__.py", line 869, in format
    return fmt.format(record)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/logging/__init__.py", line 608, in format
    record.message = record.getMessage()
  File "xxx/miniconda3/envs/m2m/lib/python3.7/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "xxx/miniconda3/envs/m2m/bin/m2m_analysis", line 8, in <module>
    sys.exit(main())
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/__main_analysis__.py", line 289, in main
    args.oog, new_arg_modelhost, args.level)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/__main_analysis__.py", line 303, in main_analysis_workflow
    run_analysis_workflow(*allargs)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/m2m_analysis/m2m_analysis_workflow.py", line 46, in run_analysis_workflow
  gml_output = graph_analysis(json_file_folder, target_folder_file, output_dir, taxon_file, taxonomy_level)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/m2m_analysis/solution_graph.py", line 58, in graph_analysis
    create_gml(json_paths, target_paths, output_dir, taxonomy_output_file)
  File "xxx/miniconda3/envs/m2m/lib/python3.7/site-packages/metage2metabo/m2m_analysis/solution_graph.py", line 116, in create_gml
    logger.warning('ERROR ', dicti["still_unprod"], ' is unproducible')
Message: 'ERROR '
Arguments: (['M_CPD__45__1823_c', 'M_HISTAMINE_c', 'M_TYRAMINE_c'], ' is unproducible')
```

An enhancement would be to drop targets that are not producible from the analysis instead of exiting in error.

**Comment by cfrioux** [comment]
@ArnaudBelcour please add these two lines here https://github.com/AuReMe/metage2metabo/blob/6fa73c2d7a014f15cb264d71af5f23fd89cd16e5/metage2metabo/m2m_analysis/solution_graph.py#L116 

```
            logger.error('Please remove these targets from the targets file and re-run m2m_analysis')
            sys.exit(1)

```

**Comment by ArnaudBelcour** [comment]
Add these lines at two places:
- just after enumeration to delete the output json files associated with unproducible targets and stop the process from going after. With this modification, it is possible to rerun `m2m_analysis` and the enumeration will be launched again on the modified target file.
- during graph solution creation to avoid creating graph solution with unproducible targets.

**Comment by ArnaudBelcour** [comment]
Fixed in #61.
